### DISCUSSION TITLE
Mixing two wav decoders: Distorted

### DISCUSSION THREAD

**Discussion by jonascant** [pschatzmann/arduino-audio-tools#1652] [Q&A] [answered]
Sorry to bother you, but I need to mix up to five wav files to one mono channel. I tried for a long time, but it is always distorted.
```
#include "AudioTools.h"
#include "AudioLibs/FileLoop.h"
#include "SdFat.h"
#include "sdios.h"

//SD Pin
const uint8_t SD_CS_PIN = 5;

#define FS_NO_GLOBALS
#define SPI_CLOCK SD_SCK_MHZ(30)
#define SD_CONFIG SdSpiConfig(SD_CS_PIN, DEDICATED_SPI, SPI_CLOCK)

SdFat32 SD;
File32 file;
File32 file1;

AudioInfo info(48000, 1, 32);

I2SStream out; 
WAVDecoder wav;


OutputMixer<int32_t> mixer(out, 2);
EncodedAudioStream encoded(&mixer, &wav); // Decoding stream
EncodedAudioStream encoded1(&mixer, &wav); // Decoding stream

StreamCopy copier(mixer, file);
StreamCopy copier1(mixer, file1);


void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Error);  

  //SD init
  if (!SD.begin(SD_CONFIG)) {
    SD.initErrorHalt(&Serial);
  }

  file = SD.open("/Jodler.wav");
  file1 = SD.open("/Baby.wav");

  // start I2S ou
  auto config_out = out.defaultConfig(TX_MODE);
  config_out.copyFrom(info); 
  config_out.port_no = 0;
  config_out.pin_bck = 15;
  config_out.pin_ws = 14;
  config_out.pin_data = 22;
  // setup I2S based on sampling rate provided by decoder
  encoded.begin();
  encoded1.begin();
  mixer.begin();
  out.begin(config_out);
}

void loop(){
  copier.copy();
  copier1.copy();
}
```

**Comment by pschatzmann** [comment, marked as answer]
Microcontrollers have only limited IO and CPU speed: The challenge is always to find out where the bottleneck is and engineer the solution accordingly.

5 * 48000 * 32 bits sounds absolutely crazy!

**Comment by jonascant** [comment]
yeah. I will bite the bullet and try to get it done on rpi4 using python... I was hoping, i could use your fantastic lib.

**Reply by pschatzmann** [reply]
You still can, but you will need to reduce the bits per sample and ev. sample rate....

ps. you could also use this library on a rpi4...

**Reply by jonascant** [reply]
i did reduce to 16000hz, 16bit with only two files. the same result. I am thinking the parallel reading of two files from sd is the issue. it plays fine if only one file is added to the mixer.

> ps. you could also use this library on a rpi4...
I have no idea how to compile the code to run on the rpi. I have only ever used arduino-cli for all compiling.
